### DISCUSSION TITLE
Support for PostgreSQL Database Attachment via SSL Mode in DuckDB

### DISCUSSION THREAD

**Discussion by BGireesh** [duckdb/duckdb#13832] [Ideas]
Summary:
Enable DuckDB to connect to PostgreSQL databases using SSL mode for enhanced security during data attachments.

Description:
Currently, DuckDB supports the attachment of external PostgreSQL databases for querying and analysis purposes. However, the existing implementation lacks support for SSL connections, which are critical for securely connecting to PostgreSQL instances, especially when dealing with sensitive or production data across networks.

Proposed Enhancement:
Add SSL mode support to the PostgreSQL attachment functionality in DuckDB. This would allow users to specify SSL-related connection parameters, enabling secure, encrypted connections between DuckDB and PostgreSQL databases.

Key Features:

	1.SSL Mode Parameter: Allow users to specify SSL mode options (e.g., require, verify-ca, verify-full) when attaching a PostgreSQL database.
	2.SSL Certificates: Support the inclusion of SSL certificate paths (CA certificate, client certificate, client key) as part of the connection string or as separate parameters.

**Comment by qvsdeveloper** [comment]
This would be great
